docs/user/lang.rst

```diff
@@ -0,0 +1,9 @@
+.. _lang:
+
+Language semantics
+==================
+
+In general, the semantics of the Scala Native language are the same as Scala on
```

Comment:
I think here is almost fair to mention memory allocation with ```stackalloc``` and ```malloc```.
And probably add a note on threading execution (i.e. not supported now)